plutus-tx/src/PlutusTx/Enum.hs

```diff
@@ -0,0 +1,17 @@
+{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}
+module PlutusTx.Enum (Enum(..)) where
+
+import           PlutusTx.Builtins
+import           Prelude           hiding (Enum (..))
+
+class Enum a where
```

Comment:
Haddocks (copying or referencing the GHC definition is OK)

Reply:
Added some comments. The whole `Enum`'s haddock contains facts about `Enum`'s connection with `Bounded` but we don't have it yet.